app/controllers/api/v1/indicators_controller.rb

```diff
@@ -5,12 +5,38 @@ def index
         indicators = Indicator.
           includes(:category, :subcategory, :model)
 
-        if params[:category]
+        if param_list(:category)
           indicators = indicators.where(
-            category_id: params[:category]
-          ).or(indicators.where(
-            subcategory_id: params[:category]
-          ))
+            category_id: param_list(:category)
+          )
+        end
+
+        if param_list(:subcategory)
+          indicators = indicators.where(
+            subcategory_id: param_list(:subcategory)
+          )
+        end
+
+        if param_list(:scenario)
+          indicator_ids = Scenario.where(id: param_list(:scenario)).
+            includes(model: :indicators).
```

Comment:
Fetch related indicators through time_series_values as the relationship between models and indicators will be removed in Kuba's PR.